### PR TITLE
[PyTorch] Correct comment on #310353

### PR DESCRIPTION
**Pull request by swolchok**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #139369
* #139363

@malfet pointed out that the version I listed was a C++ runtime version.

Differential Revision: [D65253971](https://our.internmc.facebook.com/intern/diff/D65253971/)

### CODE REVIEW

File: c10/macros/Macros.h [hunk 1]
```diff
@@ -217,8 +217,7 @@ using namespace c10::xpu;
 // Unlike C10_ALWAYS_INLINE, C10_ALWAYS_INLINE_ATTRIBUTE can be used
 // on a lambda.
 #if defined(_MSC_VER)
-// MSVC 14.39 is reasonably recent and doesn't like
-// [[msvc::forceinline]] on a lambda, so don't try to use it.
+// VS2022 doesn't accept [[msvc::forceinline]] on a lambda..
```
[the remark this answers is not in the document]
Comment (swolchok) [reply c10/macros/Macros.h:220]: maybe 14.39 is just so old godbolt doesn't have it? I don't know much about MSVC versions

[the remark this answers is not in the document]
Comment (swolchok) [reply c10/macros/Macros.h:220]: 17.9 is the MSVC version that rejects this: https://godbolt.org/z/Wz5nv386d

[the remark this answers is not in the document]
Comment (malfet) [reply c10/macros/Macros.h]: Looks like all of the versions on godbolt accept `[[forceinline]]` (And VC17.9 for some reason adds extra `[[msvc::` line entry to it, see error from you snippet `error C2416: attribute [[msvc::msvc::forceinline]] cannot be applied in this context` )

**Comment by swolchok** [comment]
let's do [[forceinline]] then

**Pull request closed without merging**